fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/AndOrValue.java

```diff
@@ -68,6 +70,8 @@ public class AndOrValue extends AbstractValue implements BooleanValue {
     @Nonnull
     private final Value rightChild;
     @Nonnull
+    private final Supplier<Iterable<? extends Value>> childrenSupplier = Suppliers.memoize(this::computeChildren);
```

Comment:
Again, please condense into `AbstractValue`

Reply:
Done, I also refactored `ValueWithChild` implementations to avoid extra allocation.